src/Sylius/Behat/Context/Setup/CartContext.php

```diff
@@ -126,8 +123,22 @@ public function iAddThisProductWithToTheCart(
         ));
     }
 
-    private function pickupCart(): string
+    /**
+     * @Given /^this (cart) has promotion applied with coupon "([^"]+)"$/
+     */
+    public function thisCartHasCouponAppliedWithCode(?string $tokenValue, string $couponCode): void
+    {
+        $tokenValue = $this->pickupCart($tokenValue);
+
+        $this->commandBus->dispatch(ApplyCouponToCart::createFromData($tokenValue, $couponCode));
+    }
+
+    private function pickupCart(?string $tokenValue = null): string
     {
+        if ($tokenValue !== null) {
```

Comment:
Why did you refactor that? IMO the previous behaviour makes more sense, when there is a method for pickuping the cart independently from passed token value and using that method only if needed.

Reply:
It was used in every method of this Context where `pickupCart` was used. I moved it here so we don't rly duplicate this code.

Reply:
As I wrote above, for me, the previous behaviour was more clear, now I am a little confused because there is method `pickupCart` and I can pass there a `$tokenValue`, so the expected behaviour is pickup the cart with given token, but it behaves differently. I would be for reverting this minor refactor.